### DISCUSSION TITLE
How to stop `rose-recursive-disassemble` from outputting CFG pointers?

### DISCUSSION THREAD

**Discussion by Cryspia** [rose-compiler/rose#193] [Q&A] [answered]
The `rose-recursive-disassemble` executable outputs pointers indicating the relationships of instructions inserted at the front of each line. This feature is indeed useful for recovering the CFG, but can cause output file size explosion when disassembling complicated binaries (for example, I tried a `gcc` binary which was around 20MB and the output file was over 1.4TB and still did not finish).

I have read the document and tried multiple arguments, but it seems that none of them work (the `--old-unparser` did work but it completely changed the output format). I would like to know that are there arguments that can remove the CFG pointers while keeping the other information format unchanged.

**Comment by matzke1** [comment, marked as answer]
`$ rose-recursive-disassemble --no-bb-cfg-arrows a.out`.

You could also use the `bat-ana` and `bat-dis` or `bat-dis-simple` tools like this:

`$ bat-ana --output=a.out.rba a.out && bat-dis --no-bb-cfg-arrows a.out.rba` 

or

`$ bat-ana -q a.out | bat-dis -q --no-bb-cfg-arrows`

The benefit of using `bat-ana` is that it creates a ROSE Binary Analysis (RBA) file on which many other ROSE tools operate and can often be faster than disassembling the specimen repeatedly for each tool.
